packages/tempus/unit_test/Tempus_UnitTest_TimeEvent.cpp

```diff
@@ -0,0 +1,1080 @@
+  TEST_COMPARE(te->indexOfNextEvent( 0), ==,  1);
+
+  //   Around last event.
+  TEST_COMPARE(te->indexOfNextEvent(2), ==,  4);
+  TEST_COMPARE(te->indexOfNextEvent(4), ==,  4);
+  TEST_COMPARE(te->indexOfNextEvent(9), ==,  4);
+
+  // Test eventInRangeIndex.
+  //   Right end.
+  TEST_COMPARE(te->eventInRangeIndex(-12.0, -10), ==, false);   // Around first event.
+  TEST_COMPARE(te->eventInRangeIndex(-12.0,  -9), ==, true );
+  TEST_COMPARE(te->eventInRangeIndex(-12.0,  -8), ==, true );
+
+  TEST_COMPARE(te->eventInRangeIndex(-4, -3), ==, false);   // Around mid event.
+  TEST_COMPARE(te->eventInRangeIndex(-4, -2), ==, true );
+  TEST_COMPARE(te->eventInRangeIndex(-4, -1), ==, true );
+
+  TEST_COMPARE(te->eventInRangeIndex(3, 3), ==, false);   // Around last event.
+  TEST_COMPARE(te->eventInRangeIndex(3, 4), ==, true );
+  TEST_COMPARE(te->eventInRangeIndex(3, 6), ==, true );
+
+  //   Left end.
+  TEST_COMPARE(te->eventInRangeIndex(-12, -7), ==, true );   // Around first event.
+  TEST_COMPARE(te->eventInRangeIndex( -9, -7), ==, true );
+  TEST_COMPARE(te->eventInRangeIndex( -8, -7), ==, false);
+
+  TEST_COMPARE(te->eventInRangeIndex(-3, 0), ==, true );   // Around mid event.
+  TEST_COMPARE(te->eventInRangeIndex(-2, 0), ==, true );
+  TEST_COMPARE(te->eventInRangeIndex(-1, 0), ==, false);
+
+  TEST_COMPARE(te->eventInRangeIndex(3, 8), ==, true );   // Around last event.
+  TEST_COMPARE(te->eventInRangeIndex(4, 8), ==, true );
+  TEST_COMPARE(te->eventInRangeIndex(5, 8), ==, false);
+
+  // Check base class defaults (functions not implemented in TimeEventListIndex).
+  TEST_COMPARE(te->isTime(1.0), ==, false);
+  TEST_COMPARE(te->timeToNextEvent(1.0), ==, te->getDefaultTime());
+  TEST_COMPARE(te->timeOfNextEvent(1.0), ==, te->getDefaultTime());
+  TEST_COMPARE(te->eventInRange(1.0, 4.0), ==, false);
+
+}
+
+
+// ************************************************************
+// ************************************************************
+TEUCHOS_UNIT_TEST(TimeEventListIndex, Construction)
+{
+  std::vector<int> testVector;
+  testVector.push_back(-2);
+  testVector.push_back( 0);
+  testVector.push_back( 7);
+  testVector.push_back( 3);
+  testVector.push_back(-5);
+
+  auto te = rcp(new Tempus::TimeEventListIndex<double>(
+                "TestName", testVector));
+
+  TEST_COMPARE(te->getName(), ==, "TestName");
+
+  auto testList = te->getIndexList();
+  TEST_COMPARE(testList.size(), ==, 5);
+  TEST_COMPARE(testList[0], ==, -5);
+  TEST_COMPARE(testList[1], ==, -2);
+  TEST_COMPARE(testList[2], ==,  0);
+  TEST_COMPARE(testList[3], ==,  3);
+  TEST_COMPARE(testList[4], ==,  7);
+
+  // Test adding a duplicate event index.
+  te->addIndex(3);
+  TEST_COMPARE(te->getIndexList().size(), ==, 5);
+  te->addIndex(1);
+  TEST_COMPARE(te->getIndexList().size(), ==, 6);
+}
+
+
+
+// ************************************************************
+// ************************************************************
```

Comment:
Also to be consistent with other tests, lets add the macros to enable individual tests. See ERK test.

Reply:
So the macro defines I think you are talking about, I have been slowly removing them.  They maybe all gone now.  I did this because Roger mentioned that we can get the same functionality via the "--test", e.g., `Tempus_UnitTest_TimeEvent.exe --test="Construction"`.  The only downside here is having better and more unique test names.  See above comment.

Reply:
learned something new today